torch/fx/experimental/sym_node.py

```diff
@@ -138,15 +138,30 @@ def compute_hint():
         # Record the FX node of the current node if we are doing translation
         # validation. They will be used for building the input assertions for
         # the translation validation problem.
-        self.fx_node = (
-            fx_node if self.shape_env._translation_validation_enabled else None
+        tx_validation_en = (
+            self.shape_env and self.shape_env._translation_validation_enabled
         )
+        self.fx_node = tx_validation_en and fx_node
 
     def with_shape_env(self, shape_env: "ShapeEnv") -> "SymNode":
         return SymNode(
             self._expr, shape_env, self.pytype, self._hint, self.constant, self.fx_node
         )
 
+    def _value_eq(self, other: "SymNode") -> bool:
+        # Purposely don't include the shape_env in the eq.
+        return (
+            self._expr == other._expr
+            and self.pytype == other.pytype
+            and self._hint == other._hint
+            and self.constant == other.constant
+            and self.fx_node == other.fx_node
```

Comment:
I actually think you probably only need expr and pytype, the rest are derived quantities that don't matter for the purpose of cache matching. Actually, pytype also probably not needed either since you're probably only ever looking at integral quantities.

Reply:
If we were going to do that I'd want to move this check over to fake_tensor.py instead of having it on SymNode. It seems kind of weird to have a function on SymNode that was specific to caching fake tensors (and not have it named something like `value_eq_for_fake_tensor_cache`)

Reply:
I think i had to use pytype for symbool previously